pkgs/development/libraries/gstreamer/gstreamermm/default.nix

```diff
@@ -1,26 +1,62 @@
-{ lib, stdenv, fetchurl, pkg-config, file, glibmm, gst_all_1, gnome }:
+{ lib
+, stdenv
+, fetchFromGitLab
+, autoreconfHook
+, perl
+, mm-common
+, pkg-config
+, file
+, glibmm
+, gst_all_1
+, nix-update-script
+}:
+
 stdenv.mkDerivation rec {
   pname = "gstreamermm";
-  version = "1.10.0";
+  version = "unstable-2022-05-03";
 
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "0q4dx9sncqbwgpzma0zvj6zssc279yl80pn8irb95qypyyggwn5y";
+  src = fetchFromGitLab {
+    # https://gitlab.gnome.org/GNOME/gstreamermm
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "gstreamermm";
+    # https://gitlab.gnome.org/GNOME/gstreamermm/-/merge_requests/5
+    rev = "89b2ece5cab32b5094686bb96426a004029a9449";
+    sha256 = "sha256-b6719eTiBJ21eNA/tgul6GuAkKF0gmsbbG2x9NAtW3s=";
   };
 
   outputs = [ "out" "dev" ];
 
-  nativeBuildInputs = [ pkg-config file ];
+  nativeBuildInputs = [
+    autoreconfHook
+    mm-common
+    pkg-config
+    file
+  ];
 
-  propagatedBuildInputs = [ glibmm gst_all_1.gst-plugins-base ];
+  propagatedBuildInputs = [
+    glibmm
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+  ];
 
   enableParallelBuilding = true;
 
+  GMMPROC_DIR = "${glibmm}/lib/glibmm-2.4/proc";
+  PERL = (perl.withPackages (p: [ p.XMLParser ])) + "/bin/perl";
+
+  # must run before autoreconfPhase
+  postPatch = ''
+    autoupdate
+    sed -i '/^autoreconf /d' autogen.sh
```

Comment:
We should just use `autogen.sh` rather than relying on `autoreconfHook`, when the build system does not work with `autoreconf`. Where did you even get `autoupdate`?

Reply:
`autoreconfHook` is preferred in nixpkgs
autoconf will set the wrong prefix

```
    Install Prefix:              /usr/local
```

and instead of fixing the prefix, i rather use autoreconfHook in the first place

autoupdate is suggested by configure
its just a warning, but should be fixed upstream

```
configure.ac:48: the top level
configure.ac:49: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:49: You should run autoupdate.
```

diff from autoupdate is

```diff
diff --git a/configure.ac b/configure.ac
index d3ded32..4f144c4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,10 +15,8 @@
 
 m4_define([GSTMM_VERSION], [1.10.0])
 
-AC_INIT([gstreamermm], [1.10.0],
-        [http://bugzilla.gnome.org/enter_bug.cgi?product=gstreamermm],
-        [gstreamermm], [http://www.gtkmm.org/])
-AC_PREREQ([2.59])
+AC_INIT([gstreamermm],[1.10.0],[http://bugzilla.gnome.org/enter_bug.cgi?product=gstreamermm],[gstreamermm],[http://www.gtkmm.org/])
+AC_PREREQ([2.71])
 
 AC_CONFIG_SRCDIR([gstreamer/gstreamermm.h])
 AC_CONFIG_AUX_DIR([build])
@@ -45,8 +43,11 @@ AC_PROG_CXX
 MM_AX_CXX_COMPILE_STDCXX_11([noext],[mandatory])
 
 AC_DISABLE_STATIC
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
+AC_REQUIRE(AC_CANONICAL_HOST)_LT_SET_OPTION([LT_INIT],[win32-dll])
+m4_warn([obsolete],[AC_LIBTOOL_WIN32_DLL: Remove this warning and the call to _LT_SET_OPTION when you
+put the 'win32-dll' option into LT_INIT's first parameter.])
+
+LT_INIT
 
 # Check for m4 and sed which are used for plug-in .hg and .ccg file generation.
 AC_CHECK_PROGS([M4], [gm4 m4], [m4])

```

Reply:
> autoconf will set the wrong prefix
> 
> ```
>     Install Prefix:              /usr/local
> ```

That does not make sense at all. AIUI, `autoconf` generates files, it does not set anything. The prefix is set by calling `configure` (in `configurePhase`, since we should run `./autogen.sh` with `NOCONFIGURE=1`).

I would say `autoreconfHook` should only be preferred when `autogen.sh` only contains `autoreconf` call and nothing else.

Either way, the following patch shows the correct prefix for me:

<details>
<summary>Patch</summary>


```patch
--- a/pkgs/development/libraries/gstreamer/gstreamermm/default.nix
+++ b/pkgs/development/libraries/gstreamer/gstreamermm/default.nix
@@ -1,7 +1,9 @@
 { lib
 , stdenv
 , fetchFromGitLab
-, autoreconfHook
+, autoconf
+, automake
+, libtool
 , perl
 , mm-common
 , pkg-config
@@ -28,7 +30,9 @@ stdenv.mkDerivation rec {
   outputs = [ "out" "dev" ];
 
   nativeBuildInputs = [
-    autoreconfHook
+    autoconf
+    automake
+    libtool
     mm-common
     pkg-config
     file
@@ -45,10 +49,7 @@ stdenv.mkDerivation rec {
   GMMPROC_DIR = "${glibmm}/lib/glibmm-2.4/proc";
   PERL = (perl.withPackages (p: [ p.XMLParser ])) + "/bin/perl";
 
-  # must run before autoreconfPhase
-  postPatch = ''
-    autoupdate
-    sed -i '/^autoreconf /d' autogen.sh
+  preConfigure = ''
     NOCONFIGURE=1 ./autogen.sh
   '';
 
```

</details>

Reply:
> ```
>     Install Prefix:              /usr/local
> ```

my bad, i had autoconf.sh in configurePhase